ServiceBusDashboard/Code/RazorUtil.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using System.Web.Routing;
+using ServiceBusDashboard.Controllers;
+
+namespace ServiceBusDashboard.Code
+{
+    public static class RazorUtil
+    {
+        public static string RenderViewToString(string controllerName, string viewName, object viewData)
+        {
+            using (var writer = new StringWriter())
+            {
+                var routeData = new RouteData();
+                routeData.Values.Add("controller", controllerName);
+                var fakeControllerContext = new ControllerContext(new HttpContextWrapper(new HttpContext(new HttpRequest(null, "http://google.com", null), new HttpResponse(null))), routeData, new FakeController());
```

Comment:
What's the purpose of the fake controller?

Reply:
This is for rendering a view regardless of where we are, either inside a controller or apicontroller or even static code.